vars/handlePipelineStepErrors.groovy

```diff
@@ -81,14 +81,19 @@ void call(Map parameters = [:], body) {
             throw ex
         }
 
-        if (config.stepParameters?.script) {
-            config.stepParameters?.script.currentBuild.result = 'UNSTABLE'
-        } else {
-            currentBuild.result = 'UNSTABLE'
+        def failureMessage = "[${STEP_NAME}] Error in step ${config.stepName} - Build result set to 'UNSTABLE'"
+        try {
+            //use new unstable feature if available: see https://jenkins.io/blog/2019/07/05/jenkins-pipeline-stage-result-visualization-improvements/
+            unstable(failureMessage)
```

Comment:
is there no better way to check if the step is available? check for plugin name and version or something? this is a little abuse of exceptions for control flow..

Reply:
That's the relation mentioned in the blog

> he core fixes for the issue are present in the following plugins, all of which require Jenkins 2.138.4 or newer:
>Pipeline: API 2.34
>Pipeline: Basic Steps 2.18 (requires a simultaneous update to Pipeline: Groovy 2.70)
>Pipeline: Graph Analysis 1.10
>Pipeline: Declarative 1.3.9
>Blue Ocean 1.17.0

Not sure if we want to check for this ...
Is there a utils function available to check for plugin availability and version?

Since we do not need to guide the user into a certain direction I think it is ok to 'misuse' the exception ...

Reply:
`JenkinsUtils#isPluginActive()` could be changed in a way that it can also verify that at least version x is installed. Not sure if it is worth the effort, just wanted to point it out so we can make up our mind.

Reply:
I agree, it is not too nice to just rely on the exception.

In light, 
that we don't need to inform the users actively about required versions in this case, 
and that this should rather be temporary and be removed at some point in time
I guess it is not really worth the effort in my eyes.